CookieLaw and enhanced privacy settings

keul opened the pull request:
See #51

hvelarde commented on CHANGES.rst:
```diff
@@ -6,6 +6,12 @@ There's a frood who really knows where his towel is.
 2.3 (unreleased)
 ^^^^^^^^^^^^^^^^
 
+- Flake8 cleanup
```
IMO, changelog entries should provide useful information for end users, developers and integrators; Flake8 fixes doesn't fall under this... but that's only MO.

keul replied:
@hvelarde no opposition about this, is not really useful.
It take me one minute for remove the entry

hvelarde commented on README.rst:
```diff
@@ -79,9 +79,27 @@ Go to the 'Site Setup' page in the Plone interface and click on the
+Social networks and privacy togheter is a thorny argument, let say that
+social media widget commonly tracks user actions and add 3rd party cookies.
+
+If privacy is something you must care about (for example: if you need to take
```
I think we should provide a link to the Cookie Law text here

hvelarde commented on sc/social/like/browser/viewlets.py:
```diff
@@ -68,4 +71,20 @@ class SocialLikesViewlet(BaseLikeViewlet):
+
+    @property
+    def render_method(self):
+        tools = getMultiAdapter((self.context, self.request),
```
can we use plone.api here?

keul replied:
This will be the ''only'' place where plone.api is used. Adding this dependency before #44 will be implemented isn't too much?

hvelarde replied:
good point; let's do that later then.

hvelarde commented on sc/social/like/plugins/facebook/browser.py:
```diff
@@ -7,6 +7,7 @@
 from sc.social.like.plugins.facebook.utils import facebook_language
 from sc.social.like.utils import get_content_image
 from sc.social.like.utils import get_language
+from urllib import quote
```
better use [urlencode](https://docs.python.org/2/library/urllib.html#urllib.urlencode) here

hvelarde commented on sc/social/like/plugins/gplus/browser.py:
```diff
@@ -3,6 +3,7 @@
 from Products.Five import BrowserView
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 from sc.social.like.utils import get_language
+from urllib import quote
```
better use [urlencode](https://docs.python.org/2/library/urllib.html#urllib.urlencode) here

hvelarde commented on sc/social/like/plugins/interfaces.py:
```diff
@@ -17,6 +17,9 @@ def metadata():
         ''' Return html render method '''
 
+    def link():
+        ''' Return html render method when high privacy is required'''
```
triple double quotes in docstrings, please

keul replied:
I found triple single quote everywhere so I respected the current code style (commonly I use triple double too)

hvelarde replied:
yes, I know, but we need to start replacing them at some point.

hvelarde commented on sc/social/like/plugins/linkedin/browser.py:
```diff
@@ -3,6 +3,7 @@
 from Products.Five import BrowserView
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 from sc.social.like.utils import get_language
+from urllib import quote
```
better use [urlencode](https://docs.python.org/2/library/urllib.html#urllib.urlencode) here

hvelarde commented on sc/social/like/plugins/pinterest/browser.py:
```diff
@@ -4,6 +4,7 @@
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 from sc.social.like.utils import get_content_image
 from sc.social.like.utils import get_language
+from urllib import quote
```
better use [urlencode](https://docs.python.org/2/library/urllib.html#urllib.urlencode) here

hvelarde commented on sc/social/like/plugins/twitter/browser.py:
```diff
@@ -5,6 +5,7 @@
 from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
 from Products.PythonScripts.standard import url_quote
 from sc.social.like.utils import get_language
+from urllib import quote
```
better use [urlencode](https://docs.python.org/2/library/urllib.html#urllib.urlencode) here

hvelarde commented on sc/social/like/upgrades/v3020/handler.py:
```diff
@@ -0,0 +1,13 @@
+
+
+def add_privacy_setting(context):
+    ''' Apply upgrade profile '''
```
triple double quotes in docstrings, please

hvelarde commented:
@keul nice work! the only thing I would love to change in the name and description of the field... I think ``privacy`` is not a good one, neither "Severe privacy".

In the future we probably want to split this feature out of this package; I've seen this implemented on a different way: some European news sites, for instance, show a small dialog when you visit their sites telling you that they are using cookies; we should do the same.

![selection_001](https://cloud.githubusercontent.com/assets/341393/8524298/b46fd684-23d0-11e5-8350-363f00da31e0.jpg)

keul commented:
@hvelarde if you can find a better name for the field I can change it! I simply didn't find any better option.

About the global warning: we use it too, but in another package (because "social" is not the only possible cookie origin) but this is a different (and complex) argument as every EU country interpreted the law in a different way.

keul commented:
Still left:
* change urllib.quote with urlencode (this needs some changes to the code)
* find a better name for the privacy field (suggestion welcome)

hvelarde commented on sc/social/like/upgrades/v3020/configure.zcml:
```diff
@@ -0,0 +1,28 @@
+    xmlns:gs="http://namespaces.zope.org/genericsetup"
+    i18n_domain="sc.social.like">
+
+    <gs:registerProfile
```
I think we also need to hide this profile

keul replied:
Yep! Right

keul commented:
@hvelarde hidden

hvelarde commented on sc/social/like/controlpanel/likes.py:
```diff
@@ -72,6 +72,19 @@ class IProvidersSchema(Interface):
         vocabulary=styles,
     )
 
+    privacy = schema.Bool(
```
what about `avoid_tracking` of `do_not_track`?

keul commented:
@hvelarde thanks for contributing, give me some days to check those changes.
IMHO "Do not track" is a good label too, :+1:

keul commented:
@hvelarde I used the do_not_track version (hoping this will be the final choice)
Also: new URL schema changes are OK.

hvelarde commented:
@keul awesome! could you please update the changelog to reflect better what was done? don't hesitate on adding a whole paragraph.

keul commented:
@hvelarde I added few words in the CHANGES file, just because I already added a new whole section in the main README. Feel free to extend/fix it!

If you can spread a new release, it will be great.

hvelarde commented:
@keul done! https://pypi.python.org/pypi/sc.social.like/2.3
